vllm/lora/punica.py

```diff
@@ -5,17 +5,35 @@
 import torch
 
 from vllm import _custom_ops as ops
+from vllm.logger import init_logger
+from vllm.lora import native_kernels
+from vllm.utils import is_cpu
+
+logger = init_logger(__name__)
+
+if is_cpu():
+    logger.warning(
+        "punica LoRA kernels on CPU will be fallback to Torch native kernels, "
+        "which will cause non-optimized performance.")
```

Comment:
nit: I think "native kernels" here is a bit misleading since the PyTorch kernels themselves are not slow.

```suggestion
        "The CPU backend does not support custom kernels for LoRA. Falling back to unoptimized PyTorch-native implementation, which may lead to performance drop.")
```